kuzzleio/kuzzle
# Allow internal cache hot reload

## What does this PR do ?

This PR allow Kuzzle to use index/collection created with Elasticsearch without restart.   
Now the method `indexCache.exists()` is able to query Elasticsearch when an index/collection does not exists and update the cache accordingly.  

The hot reload feature can also be disabled, it was necessary because the internal cache is used by the Kuzzle bootstrap at startup and at this time the Elasticsearch service is not available.  

Fix https://github.com/kuzzleio/kuzzle/issues/1111
Related https://github.com/kuzzleio/kuzzle/issues/1110

### How should this be manually tested?

1.  Download and unzip : `wget -O issue_files.zip https://gist.github.com/Aschen/872aae178e441edd8809f939988bcbdb/archive/fab0b2b420870cfb9e32e5bad972c5bdf7ae2e31.zip && unzip issue_files.zip && mv 872aae1*/* . && rmdir 872aae1* && rm issue_files.zip` 
2. Create an index + collection with ES : `node es-index.js`
3. Create a document with ES : `node es-document.js`
4. Update it with Kuzzle, you should not encounter a `Precondition failed` error

### Other changes
